sm/plugins/tedge_apt_plugin/tests/main.rs

```diff
@@ -0,0 +1,167 @@
+#[cfg(test)]
+mod apt_plugin_integration_tests {
+    use curl::easy::Easy;
+    use hamcrest2::prelude::*;
+    use serial_test::serial;
+    use std::error;
+    use std::fs::File;
+    use std::io::prelude::*;
+    use std::path::PathBuf;
+    use std::process::{Command, Stdio};
+    use std::str::FromStr;
+    use std::sync::Once;
+    use test_case::test_case;
+
+    type Result<T> = std::result::Result<T, Box<dyn error::Error>>;
+
+    const ROLLDICE: &str = "rolldice";
+    const TEDGE_APT_COMMAND: &str = "/etc/tedge/sm-plugins/apt";
+    const APT_COMMAND: &str = "/usr/bin/apt-get";
+    const ROLLDICE_URL: &str =
+        "http://ftp.br.debian.org/debian/pool/main/r/rolldice/rolldice_1.16-1+b3_amd64.deb";
+    const ROLLDICE_VERSION: &str = "1.16-1+b3";
+    const ROLLDICE_FILE_PATH: &str = "/tmp/rolldice_1.16-1+b3_amd64.deb";
+    static DOWNLOAD_ROLLDICE_BINARY: Once = Once::new();
+
+    pub fn download_rolldice_binary_once() {
+        DOWNLOAD_ROLLDICE_BINARY.call_once(|| {
+            let _ = fetch_url(ROLLDICE_URL).unwrap();
+        });
+    }
```

Comment:
As you observed earlier today this will fail due to the panic in the initialization closure and puts the static in `poisoned` state which causes any following tests to fail as well.
Maybe you could use `call_once_force()` instead to ignore it and try to reinitialize, or ideally make the closure infallible (not panicking).